frontends/web/src/routes/device/bitbox02/bitbox02.tsx

```diff
@@ -362,11 +362,12 @@ class BitBox02 extends Component<Props, State> {
     }
 
     private handleDisclaimerCheck = (event: React.SyntheticEvent) => {
+        type TAgreements = 'agreement1' | 'agreement2' | 'agreement3' | 'agreement4' | 'agreement5';
         const target = event.target as HTMLInputElement;
-        const key = target.id as 'agreement1' | 'agreement2' | 'agreement3' | 'agreement4' | 'agreement5';
-        const obj = {};
-        obj[key] = target.checked;
-        this.setState(obj);
+        const key = target.id;
+        this.setState({
+            [key as TAgreements]: target.checked
+        } as unknown as Pick<State, keyof State>);
```

Comment:
:exploding_head:

Reply:
💯  agree, but should I refactor in this PR, add a comment or do nothing?